python/pyarrow/tests/test_dataset.py

```diff
@@ -2315,6 +2315,29 @@ def test_write_dataset_partitioned(tempdir):
         partitioning=partitioning_schema)
 
 
+@pytest.mark.parquet
+@pytest.mark.pandas
+def test_write_dataset_partitioned_dict(tempdir):
+    directory = tempdir / "partitioned"
+    _ = _create_parquet_dataset_partitioned(directory)
+
+    # directory partitioning, dictionary partition columns
+    dataset = ds.dataset(
+        directory,
+        partitioning=ds.HivePartitioning.discover(infer_dictionary=True))
+    target = tempdir / 'partitioned-dir-target'
+    expected_paths = [
+        target / "a", target / "a" / "part-0.feather",
+        target / "b", target / "b" / "part-1.feather"
+    ]
+    partitioning_schema = ds.partitioning(pa.schema([
+        dataset.schema.field('part')]),
+        dictionaries=[pa.array(['a', 'b'])])
```

Comment:
Are you required to pass those dictionaries for writing? 
Based on a quick test locally, it seems to work without as well?

Reply:
Or even more, it doesn't seem to impact the result if I set different values there than the actual values in the partition column

Reply:
That's surprising; you should see errors: `No dictionary provided for dictionary field part: dictionary<values=string, indices=int32, ordered=0>` if you specify an incorrect dictionary and `Dictionary supplied for field part: dictionary<values=string, indices=int32, ordered=0> does not contain 'a'` if you specify a dictionary which doesn't include all the column's values

Reply:
For now I've updated the docstring and put a comment in the test to indicate that dictionaries are required fro parsing dict fields.